optimize/backend/src/test/java/io/camunda/optimize/service/util/ValidationHelperTest.java

```diff
@@ -0,0 +1,113 @@
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.optimize.service.util;
+
+import io.camunda.optimize.dto.optimize.query.analysis.BranchAnalysisRequestDto;
+import io.camunda.optimize.dto.optimize.query.report.ReportDataDto;
+import io.camunda.optimize.dto.optimize.query.report.single.configuration.AggregationDto;
+import io.camunda.optimize.dto.optimize.query.report.single.configuration.AggregationType;
+import io.camunda.optimize.dto.optimize.query.report.single.process.filter.FilterApplicationLevel;
+import io.camunda.optimize.dto.optimize.query.report.single.process.filter.InstanceStartDateFilterDto;
+import io.camunda.optimize.dto.optimize.query.report.single.process.filter.ProcessFilterDto;
+import io.camunda.optimize.service.exceptions.OptimizeValidationException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class ValidationHelperTest {
+
+  @Test
+  void shouldThrowExceptionWhenDtoIsInvalid() {
+    // given
+    final BranchAnalysisRequestDto dto = new BranchAnalysisRequestDto();
+
+    // when
+    final Throwable thrown = Assertions.catchThrowable(() -> ValidationHelper.validate(dto));
+
+    // then
+    Assertions.assertThat(thrown)
+        .isInstanceOf(OptimizeValidationException.class)
+        .hasMessageContaining("gateway activity id is not allowed to be empty or null");
+  }
+
+  @Test
+  void shouldThrowExceptionWhenTargetIsNull() {
+    // given
+    final String fieldName = "testField";
+    final Object target = null;
+
+    // when
+    final Throwable thrown = Assertions.catchThrowable(() -> ValidationHelper.ensureNotEmpty(fieldName, target));
+
+    // then
+    Assertions.assertThat(thrown)
+        .isInstanceOf(OptimizeValidationException.class)
+        .hasMessageContaining("testField is not allowed to be empty or null");
+  }
+
+  @Test
+  void shouldThrowExceptionWhenCollectionIsEmpty() {
+    // given
+    final String fieldName = "testCollection";
+    final List<String> collection = Collections.emptyList();
+
+    // when
+    final Throwable thrown = Assertions.catchThrowable(() -> ValidationHelper.ensureCollectionNotEmpty(fieldName, collection));
+
+    // then
+    Assertions.assertThat(thrown)
+        .isInstanceOf(OptimizeValidationException.class)
+        .hasMessageContaining("testCollection is not allowed to be empty or null");
+  }
+
+  @Test
+  void shouldValidateFiltersCorrectly() {
+    // given
+    final ProcessFilterDto<?> validFilter = new InstanceStartDateFilterDto();
+    validFilter.setFilterLevel(FilterApplicationLevel.INSTANCE);
+
+    // when
+    final Throwable thrown = Assertions.catchThrowable(() -> ValidationHelper.validateProcessFilters(List.of(validFilter)));
+
+    // then
+    Assertions.assertThat(thrown)
+        .isInstanceOf(NullPointerException.class);
```

Comment:
this is kind of not testing what the method really does. A more correct test would be to ensure that the parameter is invalid rather than null. In these cases, it would throw a validation exception rather than an NPE.

That said, I don't think that this class is worth so much of your time, so the baseline here is already an improvement even if coverage is less than 100%